node/src/blockchain/blockchain_interface.rs

```diff
@@ -297,20 +298,29 @@ where
 
                 match log_request.wait() {
                     Ok(logs) => {
+                        let logs_len = logs.len();
                         if logs
                             .iter()
                             .any(|log| log.topics.len() < 2 || log.data.0.len() > 32)
                         {
                             warning!(
                                 logger,
                                 "Invalid response from blockchain server: {:?}",
-                                logs
+                                logs.clone()
```

Comment:
Bert wonders: does the macro really take ownership here?

Reply:
It doesn't. I removed the `.clone()`